frontend/src/context/AuthProvider.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {createContext, ReactElement, useEffect, useState} from "react";
+import jwt_decode from 'jwt-decode'
+import {TOKEN_KEY} from "../models/Constants";
+import {ITokenConfig} from "../models/Connection";
+
+export interface IAuthContext {
+    token?: string,
+    config?: ITokenConfig,
```

Comment:
Why are you saving your http config in the AuthContext? You already are saving you token in a state... I would just save the token here and construct the config where i need it. Maybe have a look at axios instance